### PR TITLE
Handle session expired and more user friendly error message

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: pkg/proxy/router/router.go [hunk 1]
```diff
@@ -675,7 +705,8 @@ func NewServer(addr string, debugVarAddr string, conf *Conf) *Server {
 		return s.startAt.String()
 	}))
 
-	s.RegisterAndWait()
+	s.RegisterAndWait(true)
+	s.registerSignal()
```
Comment (c4pt0r) [inline pkg/proxy/router/router.go]: register signal after wait?
so we can't interrupt register phase?

File: pkg/proxy/router/router.go [hunk 2]
```diff
@@ -634,8 +647,25 @@ func (s *Server) RegisterAndWait() {
...
+}
+
+func (s *Server) resetTopo() {
+	log.Error("resetTopo")
+	if s.top != nil {
+		s.top.Close(s.conf.proxyId)
+	}
+
+	s.top = topo.NewTopo(s.conf.productName, s.conf.zkAddr, s.conf.f, s.conf.provider)
+	s.RegisterAndWait(false)
```
Comment (c4pt0r) [inline pkg/proxy/router/router.go]: create fence may cause error? because we've not remove old fence yet.

File: pkg/proxy/router/helper.go [hunk 3]
```diff
@@ -208,7 +208,11 @@ func getRespOpKeys(c *session) (*parser.Resp, []byte, [][]byte, error) {
 
 func filter(opstr string, keys [][]byte, c *session, timeoutSec int) (rawresp []byte, next bool, err error) {
 	if !allowOp(opstr) {
-		return nil, false, errors.Trace(fmt.Errorf("%s not allowed", opstr))
+		errmsg, err := respcoding.Marshal(fmt.Errorf("%s not allowed", opstr))
+		if err != nil {
+			log.Fatal("should never happend", opstr)
+		}
+		return errmsg, false, errors.Trace(fmt.Errorf("%s not allowed", opstr))
```
Comment (siddontang) [inline pkg/proxy/router/helper.go]: duplicate error msg generate, we can first create this error, then use in respcoding.Marshal and later return.

File: pkg/proxy/router/router.go (hunk 2 above)
Comment (ngaut) [inline pkg/proxy/router/router.go]: So there is a warning message

File: pkg/proxy/router/router.go (hunk 1 above)
Comment (ngaut) [inline pkg/proxy/router/router.go]: We will not gonna handle event before we are online

File: pkg/proxy/router/helper.go (hunk 3 above)
Comment (ngaut) [inline pkg/proxy/router/helper.go]: That's right, will fix it.

**Pull request merged**